privacyIDEA-impl/src/main/java/org/privacyidea/action/InitializePIContext.java

```diff
@@ -81,17 +84,17 @@ protected void doExecute(@NotNull ProfileRequestContext profileRequestContext, @
                 try
                 {
                     int otpLengthToInt = Integer.parseInt(otpLength);
-                    piFormContext = new PIFormContext(defaultMessage, otpFieldHint, otpLengthToInt, pollingInterval);
+                    piFormContext = new PIFormContext(defaultMessage, otpFieldHint, otpLengthToInt, pollingInterval, pollInBrowser, pollInBrowserUrl);

```

Comment:
this is duplicated, just put it after the catch

Reply:
or rather in line 98/99. Do the if-else try-catch to manipulate the variables and only initialize the PIFormContext once

Reply:
[1ef8d45](https://github.com/privacyidea/shibboleth-plugin/pull/46/commits/1ef8d4528dc5451873e4020c8dd4898695c90864)